### DISCUSSION TITLE
Different authentication methods for different identity schemas

### DISCUSSION THREAD

**Discussion by kszafran** [ory/kratos#1990] [Q&A] [answered]
Hello! I would like to ask if it's possible to enable different authentication methods for different identity schemas. For example, I might want to have a "customer" identity with only "password" authentication enabled, and an "employee" identity with some OIDC authentication only. Is that currently possible? It seems to me that's not the case.

**Comment by aeneasr** [comment, marked as answer]
That's currently not supported but it does sound like a good feature. Could you maybe elaborate on your exact use case where you need this?

**Reply by kszafran** [reply]
In our case, the system is available to different customers (companies). Some of them might want to use their own IdP to log into our system, and I wouldn't like to show "Login with X" to company Y users. Although it'll still be a problem for the login screen. Unless we have different subdomains or paths, we can't really know what login methods to present.

**Reply by kszafran** [reply]
While it's not our immediate requirement, we might need to support something like that in the future. Perhaps this could be taken care of by the frontend itself -- just hide some of the login methods depending on the user, at least on the settings page. I don't think it's feasible to keep the list of customers completely private. Especially with different subdomains/paths, that can be easily retrieved with various enumeration tools. Perhaps just presenting all possible login methods is the least bad approach...

Or if really needed, we could have a login screen similar to AWS, where you need to type the organization name 🤔 (still vulnerable to enumeration attacks, though, but less so that DNS enumeration).
